hedera-node/hedera-app/src/main/java/com/hedera/node/app/roster/schemas/V057RosterSchema.java

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.roster.schemas;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.hedera.hapi.node.state.Network;
+import com.hedera.hapi.node.state.NodeMetadata;
+import com.hedera.hapi.node.state.roster.Roster;
+import com.hedera.node.app.version.ServicesSoftwareVersion;
+import com.swirlds.platform.config.AddressBookConfig;
+import com.swirlds.platform.state.service.WritableRosterStore;
+import com.swirlds.state.lifecycle.MigrationContext;
+import com.swirlds.state.lifecycle.Schema;
+import com.swirlds.state.spi.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A restart-only schema that ensures state has a current roster if {@link AddressBookConfig#useRosterLifecycle()} is
+ * set; the logic is currently simplified to use the genesis roster from the genesis {@link Network}s returned by
+ * {@link MigrationContext#startupNetworks()}, and to promote any candidate roster in state to the current roster on
+ * an upgrade.
+ * <p>
+ * The full logic will be done in <a href="https://github.com/hashgraph/hedera-services/issues/16552">this</a> issue.
+ */
+public class V057RosterSchema extends Schema {
+    private static final Logger log = LogManager.getLogger(V057RosterSchema.class);
+
+    private static final long GENESIS_ROUND_NO = 0L;
+    private static final SemanticVersion VERSION =
+            SemanticVersion.newBuilder().major(0).minor(57).build();
+
+    /**
+     * The test to use to determine if a candidate roster may be
+     * adopted at an upgrade boundary.
+     */
+    private final Predicate<Roster> canAdopt;
+    /**
+     * The factory to use to create the writable roster store.
+     */
+    private final Function<WritableStates, WritableRosterStore> rosterStoreFactory;
+
+    public V057RosterSchema(
+            @NonNull final Predicate<Roster> canAdopt,
+            @NonNull final Function<WritableStates, WritableRosterStore> rosterStoreFactory) {
+        super(VERSION);
+        this.canAdopt = requireNonNull(canAdopt);
+        this.rosterStoreFactory = requireNonNull(rosterStoreFactory);
+    }
+
+    @Override
+    public void restart(@NonNull final MigrationContext ctx) {
+        requireNonNull(ctx);
+        if (!ctx.configuration().getConfigData(AddressBookConfig.class).useRosterLifecycle()) {
+            return;
+        }
+        final var rosterStore = rosterStoreFactory.apply(ctx.newStates());
+        final var startupNetworks = ctx.startupNetworks();
+        if (ctx.isGenesis()) {
+            setActiveRoster(GENESIS_ROUND_NO, rosterStore, startupNetworks.genesisNetworkOrThrow());
+        } else {
+            final long roundNumber = ctx.roundNumber();
```

Comment:
I think according to the proposal `roundNumber` for override and both upgrade cases should be `state round +1` e.g. [see here](https://github.com/hashgraph/hedera-services/blob/16066-tss-roster-update/platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md?plain=1#L244).

Reply:
True @MiroslavGatsanoga , great catch---fixed.